src/poll.rs

```diff
@@ -52,14 +90,21 @@ const MAX_REFCOUNT: usize = (isize::MAX) as usize;
 /// poll.poll(&mut events, None).unwrap();
 /// ```
 pub struct Poll {
-    // This type is `Send`, but not `Sync`, so ensure it's exposed as such.
-    _marker: marker::PhantomData<Cell<()>>,
-
     // Platform specific IO selector
     selector: sys::Selector,
 
     // Custom readiness queue
     readiness_queue: ReadinessQueue,
+
+    // Use an atomic to first check if a full lock will be required. This is a
+    // fast-path check for single threaded cases avoiding the extra syscall
+    lock_state: AtomicUsize,
```

Comment:
I would say that the main "problem" with using `Mutex` directly is mostly that the implementation lives in a dylib, so it can't be inlined, optimized, etc...